docs/advanced/error_model.md

```diff
@@ -7,3 +7,8 @@ title:  "Error Model"
 A ZIO value has a type parameter `E` which is the type of declared errors it can fail with. `E` only covers the errors which were specified at the outset. The same ZIO value could still throw exceptions in unforeseen ways. These unforeseen situations are called _defects_ in a ZIO program, and they lie outside E.
 
 Bringing abnormal situations from the domain of defects into that of `E` enables the compiler to help us keep a tab on error conditions throughout the application, at compile time. This helps ensure the handling of domain errors in domain specific ways. Defects, on the other hand, can creep silently to higher levels in our application, and, if they get triggered at all, their handling might eventually be in more general ways.
+
+## Lossless Error Model
+Zio holds onto errors, that would otherwise be lost, using `try finally`. If the `try` block throws an exception, and the `finally` block throws an exception as well, then, if these are caught at a higher level, only the finalizer's exception will be caught normally, not the exception from the try block.
```

Comment:
It would be considered a defect for `finalizer` to throw, which means that the fiber will be terminated (and all errors reported); `sandbox` or `foldCauseM` could be used to "catch" the full `Cause`, however, including the defect in the finalizer. 

Does that make sense?